modules/EvseV2G/iso_server.cpp

```diff
@@ -1329,12 +1334,24 @@ static enum v2g_event handle_iso_authorization(struct v2g_connection* conn) {
     }
     res->EVSEProcessing = (iso1EVSEProcessingType)conn->ctx->evse_v2g_data.evse_processing[PHASE_AUTH];
 
+    if (conn->ctx->evse_v2g_data.evse_processing[PHASE_AUTH] != iso1EVSEProcessingType_Finished) {
+        if (((is_payment_option_contract == false) && (conn->ctx->session.auth_timeout_eim == 0)) ||
+            ((is_payment_option_contract == true) && (conn->ctx->session.auth_timeout_pnc == 0))) {
+            dlog(DLOG_LEVEL_DEBUG, "Waiting for authorization forever!");
+        } else if ((time(NULL) - conn->ctx->session.auth_start_timeout) >=
+                   (is_payment_option_contract ? (time_t)conn->ctx->session.auth_timeout_pnc
+                                               : (time_t)conn->ctx->session.auth_timeout_eim)) {
```

Comment:
I would advise to use `clock_gettime` using the monotonic clock here, otherwise when the system time changes, this will lead to hard to debug bugs

Reply:
I do not entirely understand your point.. when would the system time change? In the period between start of counting (e.g PaymentDetails in PnC) and the check in Authorization? Can you describe more the corner case?